File: .claude/api docs/on_provizyon_yeni/Python/cancel.py
# Mağaza bilgileri
merchant_id   = 'MAGAZA_NO';
merchant_key  = b'XXXXXXXXXXX';
merchant_salt = b'YYYYYYYYYYY';

# Kullanıcı bilgileri
user_ip ='';

# İşlem bilgileri
reference_no = 'AXXXXXXXXXXXXXXXX';
merchant_oid = 'TESTPREAUTHXXX';
client_lang  = 'tr';
test_mode    = 0;


 hash_str = merchant_id + user_ip + merchant_oid +  reference_no
paytr_token = base64.b64encode(hmac.new(merchant_key, hash_str.encode() + merchant_salt, hashlib.sha256).digest())

 context = {
 
        'auth_type':'capture',
        'merchant_id': merchant_id,
        'client_ip': user_ip,
        'merchant_oid': merchant_oid,
        'test_failed': test_mode,
        'reference_no' :reference_no,
        'paytr_token': paytr_token.decode(),
        'client_lang': client_lang
    } 



 return render(request, 'home.html', context)

File: .claude/api docs/on_provizyon_yeni/PHP/cancel.php
<?php

// Mağaza bilgileri
$merchant_id   = 'MAGAZA_NO';
$merchant_key  = 'XXXXXXXXXXX';
$merchant_salt = 'YYYYYYYYYYY';

// Kullanıcı bilgileri
if( isset( $_SERVER["HTTP_CLIENT_IP"] ) ) {
    $ip = $_SERVER["HTTP_CLIENT_IP"];
} elseif( isset( $_SERVER["HTTP_X_FORWARDED_FOR"] ) ) {
    $ip = $_SERVER["HTTP_X_FORWARDED_FOR"];
} else {
    $ip = $_SERVER["REMOTE_ADDR"];
}

// İşlem bilgileri
$reference_no = 'AXXXXXXXXXXXXXXXX';
$merchant_oid = 'TESTPREAUTHXXX';
$client_lang  = 'tr';
$test_mode    = 0;

$hash_str    = $merchant_id . $ip . $merchant_oid . $reference_no;
$paytr_token = base64_encode(hash_hmac('sha256', $hash_str . $merchant_salt, $merchant_key, TRUE));

$payload = [
    'auth_type'    => 'cancel',
    'paytr_token'  => $paytr_token,
    'merchant_id'  => $merchant_id,
    'merchant_oid' => $merchant_oid,
    'reference_no' => $reference_no,
    'client_ip'    => $ip,
    'client_lang'  => $client_lang,
    'test_failed'  => $test_mode
];

$ch = curl_init();
curl_setopt($ch, CURLOPT_URL, 'https://www.paytr.com/odeme/auth');
curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);
curl_setopt($ch, CURLOPT_POST, 1);
curl_setopt($ch, CURLOPT_POSTFIELDS, $payload);
curl_setopt($ch, CURLOPT_FRESH_CONNECT, TRUE);
curl_setopt($ch, CURLOPT_TIMEOUT, 20);

$result = @curl_exec($ch);
curl_close($ch);

echo $result;